commit 411b321f0ff5856a244660997678957013ca4644
Author: Vasily Khoruzhick <anarsoul@gmail.com>
Date:   Thu Nov 8 20:31:35 2018 -0800

    ASoC: sun8i-codec: fix crash on module removal
    
    [ Upstream commit 278df5e5527b633f4882f1680ad58b62a7c07bfe ]
    
    drvdata is actually sun8i_codec, not snd_soc_card, so it crashes
    when calling snd_soc_card_get_drvdata().
    
    Drop card and scodec vars anyway since we don't need to
    disable/unprepare clocks - it's already done by calling
    runtime_suspend()
    
    Drop clk_disable_unprepare() calls for the same reason.
    
    Fixes: 36c684936fae7 ("ASoC: Add sun8i digital audio codec")
    Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
    Acked-by: Maxime Ripard <maxime.ripard@bootlin.com>
    Signed-off-by: Mark Brown <broonie@kernel.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/sound/soc/sunxi/sun8i-codec.c b/sound/soc/sunxi/sun8i-codec.c
index fb37dd927e33..bf615fa16dc8 100644
--- a/sound/soc/sunxi/sun8i-codec.c
+++ b/sound/soc/sunxi/sun8i-codec.c
@@ -589,16 +589,10 @@ static int sun8i_codec_probe(struct platform_device *pdev)
 
 static int sun8i_codec_remove(struct platform_device *pdev)
 {
-	struct snd_soc_card *card = platform_get_drvdata(pdev);
-	struct sun8i_codec *scodec = snd_soc_card_get_drvdata(card);
-
 	pm_runtime_disable(&pdev->dev);
 	if (!pm_runtime_status_suspended(&pdev->dev))
 		sun8i_codec_runtime_suspend(&pdev->dev);
 
-	clk_disable_unprepare(scodec->clk_module);
-	clk_disable_unprepare(scodec->clk_bus);
-
 	return 0;
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1032_411b321f0ff5856a244660997678957013ca4644.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1032_411b321f0ff5856a244660997678957013ca4644.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
